### DISCUSSION TITLE
Conditional platform target compilation

### DISCUSSION THREAD

**Discussion by tmpfs** [fzyzcjy/flutter_rust_bridge#1259] [General]
I want to compile some code specific for MacOS but it appears that the generated API ignores the conditional statements which make sense as the generated bindings are for all platforms.

However, this is problematic for my build process as this code really shouldn't be compiled in for other platforms. 

Is there an idiomatic way to handle this with FRB?

**Comment by tmpfs** [comment]
The only way it seems to do this currently is to stub functions on other platforms with `unimplemented!()` function bodies.

It would great if there is a more streamlined solution for this!

**Reply by fzyzcjy** [reply]
Yes I think so, currently you can create `unimplemented`.

One way is to modify such that it generates conditionally, but as you have said, this can be quite hard because generated .h can be used for multi platforms.

Another way is that, maybe we can realize the conditional compilation, and generate stub inside frb generated code. 
As for how to realize conditional flag - maybe just read it, and add a `not(...)` around it to invert it?
